addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlParser.java

```diff
@@ -152,15 +152,19 @@ public void importFile(String filePath) throws IOException {
     public void parse(String schema) {
         if (syncParse) {
             fingerprint();
-            generate(schema);
+            if (param.getQueryGenEnabled()) {
```

Comment:
Maybe show some info dialogue if not enabled? As is now it seems that nothing happens when importing with the generation disabled.

Reply:
We should try the introspection query (or a subset of it) even in the case they import a spec and raise an alert if it is supported. That way it should show up in the sites tree. +1 to handle it in another PR.